analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -82,45 +83,62 @@ public abstract class UtilityAnalyzerBase<TSyntaxKind, TMessage> : UtilityAnalyz
 
         protected virtual bool AnalyzeUnchangedFiles => false;
 
-        protected virtual IEnumerable<TMessage> CreateAnalysisMessages(CompilationAnalysisContext c) => Enumerable.Empty<TMessage>();
+        protected virtual IEnumerable<TMessage> CreateAnalysisMessages(Compilation compilation) => Enumerable.Empty<TMessage>();
 
         protected UtilityAnalyzerBase(string diagnosticId, string title) : base(diagnosticId, title) { }
 
         protected sealed override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterCompilationAction(c =>
+            context.RegisterCompilationStartAction(startContext =>
+            {
+                ReadParameters(context, startContext.Options, startContext.Compilation);
+                if (!IsAnalyzerEnabled)
+                {
+                    return;
+                }
+
+                ConcurrentStack<TMessage> treeMessages = new();
+                SonarAnalysisContext.TryGetValueDelegate<ProjectConfigReader> tryGetValue = startContext.TryGetValue;
```

Comment:
I'd remove this, it still takes permanent dependency on `startContext`, it's just more hidden and indirect

Reply:
..the `TryGetValueDelegate` will get deleted anyway

Reply:
> I'd remove this, it still takes permanent dependency on startContext

You are probably right. Are we blocked by #6532 and should wait for it to be merged?